notion-server/src/ethereum.rs

```diff
@@ -1,4 +1,77 @@
+use base64;
 use std::collections::HashMap;
+use std::convert::TryFrom;
+use snafu::{Snafu, ResultExt, Backtrace};
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    Decode {
+        backtrace: Backtrace,
+        source: base64::DecodeError,
+    },
+    OutOfBounds {
+        message: String,
+    }
+}
+
+pub mod interface_args {
+    #[derive(Debug, Default)]
+    pub struct GeneralSimulationState {
+        pub num_execution_environments: u32,
+        pub num_shard_chains: u32,
+    }
+
+    #[derive(Debug, Default)]
+    pub struct ExecutionEnvironment {
+        pub base64_encoded_wasm_code: String,
```

Comment:
I think I'd prefer to store this as a `Vec<u8>` even in the argument, and implement the conversion to/from base64 in the serialization code. Serde has [an attribute](https://serde.rs/field-attrs.html#with) for exactly that.

Not a blocking issue for the PR, just a suggestion.

Reply:
I'm on board for that if we'd rather do the serialization in the web handlers.  I think that might be better / would keep the `EthereumSimulation` interface cleaner and more flexible, so sounds good to me.

Reply:
> removing the need for a 3rd intermediate struct.

I'm not suggesting a third intermediate struct. I'm suggesting deriving `Deserialize` directly on this struct, and using serde's `with` annotation.

I'm okay with leaving it for now though.